Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -1923,13 +1923,12 @@ void TextureCacheBase::FlushEFBCopy(TCacheEntry* entry)
   WriteEFBCopyToRAM(dst, entry->pending_efb_copy_width, entry->pending_efb_copy_height,
                     entry->memory_stride, std::move(entry->pending_efb_copy));
 
-  // If the EFB copy was invalidated (e.g. the bloom case mentioned in InvalidateTexture),
-  // now is the time to clean up the TCacheEntry. In which case, we don't need to compute
-  // the new hash of the RAM copy.
+  // If the EFB copy was invalidated (e.g. the bloom case mentioned in InvalidateTexture), now is
+  // the time to clean up the TCacheEntry. In which case, we don't need to compute the new hash of
+  // the RAM copy. But we need to clean up the TCacheEntry, as InvalidateTexture doesn't free it.
   if (entry->pending_efb_copy_invalidated)
   {
-    auto config = entry->texture->GetConfig();
-    texture_pool.emplace(config, TexPoolEntry(std::move(entry->texture)));
+    delete entry;
```

Comment:
The texture is already returned to the pool in InvalidateTexture (I figured it'd be better to return it as soon as possible, rather than keeping it until the copy is flushed, when it won't be used). The TCacheEntry is just kept around to know the address where the deferred copy should be written to.